unitypack/asset.py

```diff
@@ -257,23 +257,31 @@ def add_object(self, type_id):
 
 		return obj
 
-	def add2ab(self, path, path_id, file_id=0):
+	def add2ab(self, path, path_id, file_id=0, preloads=None):
 		"Add a path -> ObjectPointer connection into the AssetBundle object"
 
 		if self.objects[1].type != 'AssetBundle':
 			raise ValueError('AssetBundle object not found. Is this a Scene bundle?')
 
 		ab = self.objects[1].contents
-		ab_index = len(ab['m_Container'])
 		preload_index = len(ab['m_PreloadTable'])
 
 		ptr = ObjectPointer(None, self) # type argument is unused
 		ptr.file_id = file_id
 		ptr.path_id = path_id
 
+		if preloads is None:
+			# assume that this is a single object preload, and add it to the table
+			# TODO: maybe we could also auto collect dependencies for gameobjects, etc.?
```

Comment:
If we do add auto dependency collection, I think that should probably be in a higher level wrapper function that calls this one, to keep this one simple.